pkg/webhook/resources/virtualmachine/mutator.go

```diff
@@ -310,8 +317,6 @@ func generateMemoryPatch(vm *kubevirtv1.VirtualMachine, mem *resource.Quantity,
 
 	if vm.Spec.Template.Spec.Domain.Memory == nil {
 		patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/template/spec/domain/memory", "value": {"guest":"%s"}}`, &guestMemory))
-	} else {
-		patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/template/spec/domain/memory/guest", "value": "%s"}`, &guestMemory))
```

Comment:
this change might break existing VMs.

1. for `legacy` mode VM, the guest is decided by `domain.resources`, UI sets `domain.resources`, mutator patchs guest per reserved label, always replace the guest vm;

2. for `hot-plug` mode vm, the guest is the source of truth

without an annotation/label on VM, we can't identify the vm's mode

with this PR, if we change legacy VM, expand the memory from `domian.resources`, the `guest` stucks on old value, esp. the `rancher manager` -> docker machine interface.

Reply:
Can this done in 1.7? Otherwise, the change and QA scope will be too big. For 1.6, hotplug updates can only be done from Harvester UI. For Rancher integration, the mutator retains the logic to fill in the guest resources if they aren't provided in the request.

Reply:
What I am thinking is, the v1.6.0 hot-plug mode vm needs to have a label;  for any VM without this lable, fallback to legacy mode.

Reply:
Rancher integration will affect both legacy and hotplug VMs. If a Rancher integration user enabled hotplug on an existing legacy VM, then the guest auto-configuration would stop working. If possible, we should find a better way to make this a more consistent UX. Up till now, Rancher integration users don't have to worry about domain resource topology configuration. Either it stays that way or we add new UI elements to Rancher. The latter increases the scope but we can have a separate discussion on timeline and milestone.